cmd/main.go

```diff
@@ -75,7 +96,10 @@ func main() {
 	}
 	defer nc.Close()
 
-	mqttClient := connectToMQTTBroker(cfg.Agent.MQTT.URL, cfg.Agent.Thing.ID, cfg.Agent.Thing.Key, logger)
+	mqttClient, err := connectToMQTTBroker(cfg.Agent.MQTT, logger)
+	if err != nil {
+		log.Fatalf(err.Error())
```

Comment:
You should use our logger, you have it imported already and error package for error wrapping.